### PR TITLE
Feature/ipld verifier

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
Add ipld verifier.

### CODE REVIEW

File: core/storage/car/car.cpp [hunk 1]
```diff
@@ -58,31 +58,37 @@ namespace fc::storage::car {
...
     for (auto &root : roots) {
-      OUTCOME_TRY(walker.recursiveAll(root));
+      Traverser traverser{store, root};
+      OUTCOME_TRY(visited, traverser.traverseAll());
+      for (auto &cid : visited) {
+        cids.insert(cid);
+      }
     }
-    return makeCar(store, roots, walker);
+    return makeCar(store, roots, std::vector<CID>(cids.begin(), cids.end()));
```
Comment (turuslan) [inline core/storage/car/car.cpp:79]:
```suggestion
    return makeCar(store, roots, {cids.begin(), cids.end()});
```

File: core/storage/car/car.cpp [hunk 2]
```diff
@@ -58,31 +58,37 @@ namespace fc::storage::car {
...
   outcome::result<Buffer> makeCar(Ipld &store, const std::vector<CID> &roots) {
-    Walker walker{store};
+    std::set<CID> cids;
     for (auto &root : roots) {
-      OUTCOME_TRY(walker.recursiveAll(root));
+      Traverser traverser{store, root};
+      OUTCOME_TRY(visited, traverser.traverseAll());
+      for (auto &cid : visited) {
+        cids.insert(cid);
+      }
```
Comment (turuslan) [inline core/storage/car/car.cpp:77]:
```suggestion
      cids.insert(visited.begin(), visited.end());
```

Comment (Alexey-N-Chernyshov) [reply core/storage/car/car.cpp]: done

File: core/storage/car/car.cpp (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply core/storage/car/car.cpp]: done

File: core/markets/retrieval/client/impl/retrieval_client_error.cpp [hunk 3]
```diff
@@ -26,6 +26,8 @@ OUTCOME_CPP_DEFINE_CATEGORY(fc::markets::retrieval::client,
              "requests";
     case RetrievalClientError::kBadPaymentRequestTooMuch:
       return "RetrievalClient: too much money requested for bytes sent";
+    case RetrievalClientError::kBlockCidParseError:
+      return "RetrievalClient: block cid parse error";
   }
 
   return "RetrievalClient: unknown error";
```
Comment (artyom-yurin) [inline core/markets/retrieval/client/impl/retrieval_client_error.cpp:33]:
```suggestion
      return "RetrievalClient: block cid parse error";
    default:
      return "RetrievalClient: unknown error";
  }
```

File: core/storage/ipld/traverser.cpp [hunk 4]
```diff
@@ -54,53 +54,70 @@ namespace fc::storage::ipld::walker {
...
+OUTCOME_CPP_DEFINE_CATEGORY(fc::storage::ipld::traverser, TraverserError, e) {
+  using fc::storage::ipld::traverser::TraverserError;
+
+  switch (e) {
+    case TraverserError::kTraverseCompleted:
+      return "Traverser: blocks already completed";
   }
-}  // namespace fc::storage::ipld::walker
+
+  return "Traverser: unknown error";
```
Comment (artyom-yurin) [inline core/storage/ipld/traverser.cpp:122]: Same here

File: core/storage/ipld/verifier.cpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+OUTCOME_CPP_DEFINE_CATEGORY(fc::storage::ipld::verifier, VerifierError, e) {
+  using fc::storage::ipld::verifier::VerifierError;
+
+  switch (e) {
+    case VerifierError::kUnexpectedCid:
+      return "IPLD Block Verifier: unexpected CID encountered";
+  }
+
+  return "IPLD Block Verifier: unknown error";
```
Comment (artyom-yurin) [inline core/storage/ipld/verifier.cpp:34]:
```suggestion
    deafult:
        return "IPLD Block Verifier: unknown error";
  }
```

**Pull request merged**